siege_utilities/political/readers.py

```
@@ -0,0 +1,236 @@
+        clauses.append("plan_type = %s")
+        params.append(plan_type)
+    if active_only:
+        clauses.append("effective_to_date IS NULL")
+
+    where = ("WHERE " + " AND ".join(clauses)) if clauses else ""
+    sql = f"SELECT * FROM {_qualified('redistricting_plans')} {where} ORDER BY effective_from_date DESC"
+    return engine.query(sql, params=params)
+
+
+def plan_district_assignments(
+    engine: Any,
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `engine`

(ANN401)